core/src/main/java/bio/terra/pearl/core/service/kit/KitStatusEvent.java

```diff
@@ -21,6 +21,11 @@ public static KitStatusEvent newInstance(KitRequest kitRequest, KitRequestStatus
                 .kitRequest(kitRequest)
                 .priorStatus(priorStatus)
                     .build();
+        } else if (KitRequestStatus.RECEIVED.equals(kitRequest.getStatus())) {
+            return KitReceivedEvent.builder()
+                .kitRequest(kitRequest)
+                .priorStatus(priorStatus)
+                    .build();
```

Comment:
The boilerplate code here is now repeated three times. Offhand I could not think of an elegant alternative, but I am open to suggestions, or I can propose something if this just feels bad.